lib/sqlite3/fork_safety.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require "weakref"
+
+# based on Rails's active_support/fork_tracker.rb
+module SQLite3
+  module ForkSafety
+    module CoreExt
+      def _fork
```

Comment:
This is only going to work in Ruby 3.1+ so we'll need to remember to drop support for Ruby 3.0 (which it's time to do, anyway).

Reply:
Done in an additional commit.